contracts/msig-launcher/src/contract.rs

```diff
@@ -34,216 +32,119 @@ pub fn instantiate(
-                    label,
-                },
-                0,
-            )))
-        }
+        } => execute_instantiate(
+            deps,
+            env,
+            info,
+            name,
+            description,
+            image_url,
+            max_voting_period,
+            members,
+        ),
     }
 }
 
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
-        QueryMsg::MSig { label } => to_json_binary(&MSIG.load(deps.storage, label)?),
-        QueryMsg::CodeIds {} => to_json_binary(&MSIG_CODE_IDS.load(deps.storage)?),
-    }
-}
-
-#[cfg_attr(not(feature = "library"), entry_point)]
-pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
-    let mut resp = Response::default();
-    let code_ids = MSIG_CODE_IDS.load(deps.storage)?;
-    let (label, sender) = PENDING_MSIG.load(deps.storage)?;
-    PENDING_MSIG.remove(deps.storage);
-
-    let mut builder = MSigBuilder::new(sender);
-
-    match msg.result {
-        SubMsgResult::Ok(result) => {
-            for event in result.events {
-                // Look for instantiate type
-                if event.ty == "instantiate" {
-                    // Look for both addr and code id
-                    let mut address = None;
-                    let mut code_id = None;
-                    for attr in event.attributes {
-                        if attr.key == "_contract_address" {
-                            address = Some(attr.value);
-                        } else if attr.key == "code_id" {
-                            code_id = Some(attr.value);
-                        }
-                    }
-
-                    // If both are found, match them into their appropriate address
-                    if let Some((address, code_id)) = address.zip(code_id) {
-                        builder.set_contract(
-                            &code_ids,
-                            code_id.parse::<u64>().unwrap(),
-                            address,
-                        )?;
-                    }
+        QueryMsg::MSigs { pagination } => {
+            let limit = pagination
+                .limit
+                .unwrap_or(PAGINATION_DEFAULT)
+                .min(PAGINATION_LIMIT) as usize;
+
+            let mut results = vec![];
+            let mut page = None;
+
+            // We add a +1 for the next page to get
+            for (i, res) in MSIG
+                .range(
+                    deps.storage,
+                    Some(Bound::inclusive((
+                        pagination.user,
+                        pagination.start_at.unwrap_or(0),
+                    ))),
+                    None,
+                    Order::Ascending,
+                )
+                .take(limit + 1)
+                .enumerate()
+            {
+                let ((_, height), result) = res?;
+                results.push(result);
```

Comment:
`limit`th one shouldn't be in `results` tho - you fetch it only to get the next cursor, right?

Reply:
That's a bug, should be resolved now